flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/NettyMessage.java

```diff
@@ -584,4 +587,60 @@ static CloseRequest readFrom(@SuppressWarnings("unused") ByteBuf buffer) throws
 			return new CloseRequest();
 		}
 	}
+
+	static class AddCredit extends NettyMessage {
+
+		private static final byte ID = 6;
+
+		final ResultPartitionID partitionId;
+
+		final int credit;
+
+		final InputChannelID receiverId;
+
+		AddCredit(ResultPartitionID partitionId, int credit, InputChannelID receiverId) {
+			this.partitionId = partitionId;
+			this.credit = credit;
```

Comment:
similar for the `PartitionRequest` message, but since this is used for the non-credit based `PartitionRequestClientHandler` for now, we need to make an exception and also allow a value of `0` there

Reply:
I think it should be `checkArgument(credit > 0)`?
I will add this check in `AddCredit` here and the `PartitionRequest` will be added in the last PR to avoid temporary value 0, what do you think?

Reply:
oh, yes, you're right - I meant `>= 1` of course

regarding `PartitionRequest`, this makes sense